src/main/java/com/google/api/generator/gapic/composer/common/AbstractServiceClientTestClassComposer.java

```diff
@@ -96,15 +96,18 @@
   private static final String MOCK_SERVICE_VAR_NAME_PATTERN = "mock%s";
   private static final String PAGED_RESPONSE_TYPE_NAME_PATTERN = "%sPagedResponse";
 
-  private final TypeStore fixedTypeStore;
-  private final ClassNames classNames;
-  private final AnnotationNode testAnnotation;
-
-  protected AbstractServiceClientTestClassComposer(
-      TypeStore fixedTypeStore, ClassNames classNames) {
-    this.fixedTypeStore = fixedTypeStore;
-    this.classNames = classNames;
-    this.testAnnotation = AnnotationNode.withType(fixedTypeStore.get("Test"));
+  protected static final TypeStore FIXED_TYPESTORE = createStaticTypes();
+  protected static final AnnotationNode TEST_ANNOTATION =
+        AnnotationNode.withType(FIXED_TYPESTORE.get("Test"));
+
+  private final TransportContext transportContext;
+
+  protected AbstractServiceClientTestClassComposer(TransportContext transportContext) {
+    this.transportContext = transportContext;
+  }
+
+  public TransportContext getTransportContext() {
```

Comment:
Should this be part of ClassComposer?

Reply:
Yep, if you are fine with adding it to the general interface I'm more than  happy to do it!

Reply:
There are many transport-unaware composers, which will make this method not-needed for them. The possible solutions would be adding something `interface TransportAwareClassComposer extends ClassComposer` with the new method. WDYT? I'll leave it unchanged for this PR, and can make this change in a subsequent PR as it would be a trivial one.

Reply:
Subsequent PR SGTM. If it's not needed in most composers, feel free to skip.